ui/desktop/tests/acceptance/projects/targets/index-test.js

```diff
@@ -290,4 +297,120 @@ module('Acceptance | projects | targets | index', function (hooks) {
     assert.dom(ROSE_DIALOG_RETRY_BUTTON).hasText('Retry');
     assert.dom(ROSE_DIALOG_CANCEL_BUTTON).hasText('Cancel');
   });
+
+  test('user can open sessions flyout when target has active or pending sessions', async function (assert) {
+    assert.expect(4);
+    await visit(urls.targets);
+
+    assert
+      .dom(
+        `[data-test-targets-sessions-flyout-button="${instances.target.id}"]`,
+      )
+      .exists();
+
+    await click(
+      `[data-test-targets-sessions-flyout-button="${instances.target.id}"]`,
+    );
+
+    assert.dom(SESSIONS_FLYOUT).exists();
+    assert.dom(SESSIONS_FLYOUT_TITLE).includesText(`Active sessions for`);
+
+    await click(SESSIONS_FLYOUT_CLOSE_BUTTON);
+
+    assert.dom(SESSIONS_FLYOUT).doesNotExist();
+  });
+
+  test('user can cancel a session from inside target sessions flyout', async function (assert) {
+    assert.expect(5);
+    this.stubClientDaemonSearch(
+      'targets',
+      'sessions',
+      'targets',
+      'targets',
+      '',
+      'targets',
+    );
+
+    await visit(urls.targets);
+    await click(
+      `[data-test-targets-sessions-flyout-button="${instances.target.id}"]`,
+    );
+
+    assert.dom(SESSIONS_FLYOUT).exists();
+    assert
+      .dom(`[data-test-session-flyout-cancel-button="${instances.session.id}"]`)
+      .exists();
+
+    await click(
+      `[data-test-session-flyout-cancel-button="${instances.session.id}"]`,
+    );
+
+    assert.dom(SESSIONS_FLYOUT).doesNotExist();
+    assert
+      .dom(
+        `[data-test-targets-sessions-flyout-button="${instances.target.id}"]`,
+      )
+      .doesNotExist();
+    assert.strictEqual(currentURL(), urls.targets);
+  });
+
+  test('user cannot cancel a session from inside target sessions flyout without permissions', async function (assert) {
+    assert.expect(4);
```

Comment:
none blocking comment: totally fine if you want to use it but letting you at the current config, `assert.expect` is not need anymore unless you are using nested assertions, or assertions within a function.

Reply:
ahh right, thanks for catching that!